tests/regressions/execution_tree/replace_names_374.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2018 Parsa Amini
+// Copyright (c) 2018 Hartmut Kaiser
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+// Fixing #374: Redefining a PhySL function in Python is not possible
+
+#include <phylanx/phylanx.hpp>
+
+#include <hpx/hpx_init.hpp>
+#include <hpx/include/iostreams.hpp>
+#include <hpx/util/lightweight_test.hpp>
+
+#include <string>
+#include <sstream>
+
+std::string const code = R"(
+    define(ultimate_answer, 42)
+    debug(ultimate_answer)
+    define(ultimate_answer, block(
+        debug(42)
+    ))
+    ultimate_answer()
+)";
```

Comment:
May I suggest to let the different instances of `ultimate_answer` actually return different things?